pkg/proto/transactions.go

```diff
@@ -20,25 +20,30 @@ type TransactionType byte
-	IssueTransaction                                      // 3 - Issue transaction
-	TransferTransaction                                   // 4 - Transfer transaction
-	ReissueTransaction                                    // 5 - Reissue transaction
-	BurnTransaction                                       // 6 - Burn transaction
-	ExchangeTransaction                                   // 7 - Exchange transaction
-	LeaseTransaction                                      // 8 - Lease transaction
-	LeaseCancelTransaction                                // 9 - LeaseCancel transaction
-	CreateAliasTransaction                                // 10 - CreateAlias transaction
-	MassTransferTransaction                               // 11 - MassTransfer transaction
-	DataTransaction                                       // 12 - Data transaction
-	SetScriptTransaction                                  // 13 - SetScript transaction
-	SponsorshipTransaction                                // 14 - Sponsorship transaction
-	SetAssetScriptTransaction                             // 15 - SetAssetScript transaction
-	InvokeScriptTransaction                               // 16 - InvokeScript transaction
-	UpdateAssetInfoTransaction                            // 17 - UpdateAssetInfoTransaction
+	GenesisTransaction          TransactionType = iota + 1 // 1 - Genesis transaction
+	PaymentTransaction                                     // 2 - Payment transaction
+	IssueTransaction                                       // 3 - Issue transaction
+	TransferTransaction                                    // 4 - Transfer transaction
+	ReissueTransaction                                     // 5 - Reissue transaction
+	BurnTransaction                                        // 6 - Burn transaction
+	ExchangeTransaction                                    // 7 - Exchange transaction
+	LeaseTransaction                                       // 8 - Lease transaction
+	LeaseCancelTransaction                                 // 9 - LeaseCancel transaction
+	CreateAliasTransaction                                 // 10 - CreateAlias transaction
+	MassTransferTransaction                                // 11 - MassTransfer transaction
+	DataTransaction                                        // 12 - Data transaction
+	SetScriptTransaction                                   // 13 - SetScript transaction
+	SponsorshipTransaction                                 // 14 - Sponsorship transaction
+	SetAssetScriptTransaction                              // 15 - SetAssetScript transaction
+	InvokeScriptTransaction                                // 16 - InvokeScript transaction
+	UpdateAssetInfoTransaction                             // 17 - UpdateAssetInfoTransaction
+	EthereumMetamaskTransaction                            // 18 - EthereumMetamaskTransaction is a transaction which received from metamask
 )
 
+func (tt TransactionType) BinarySize() int {
```

Comment:
Looks like this method is not required, because of incorrect assumption that the binary size of eth transaction prepended with type byte.